calliope/config/defaults.yaml

```diff
@@ -44,6 +44,7 @@ default_tech:
         primary_carrier_in: false  # Selects the primary input carrier to associate with costs & constraints (if multiple primary input carriers are assigned)
         primary_carrier_out: false  # Selects the primary output carrier to associate with costs & constraints (if multiple primary output carriers are assigned)
     constraints:
+        binary_operation: false  # if True, carrier_prod and carrier_con cannot both occur in the same timestep
```

Comment:
I wonder if binary operation is the right name to use, since it is quite generic. Is this future-proofing as it might encompass additional features? Or should it be called something like `no_concurrent_prod_con` which is more descriptive? Finally: should there be clearer documentation on which constraints turn on MILP and thus dramatically increase solution time -- possibly even ModelWarning that says "The following constraints you activated are forcing mixed-integer and may affect solution time: ...."?

Reply:
Re 2: Sounds sensible to use the subsets, and that could be in `preprocess.checks.check_final` which happens after all sets are built?